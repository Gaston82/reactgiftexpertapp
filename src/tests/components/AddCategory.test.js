import React from "react";
import "@testing-library/jest-dom";
import { AddCategory } from "../../components/AddCategory";
import { shallow } from "enzyme";

describe("Testing AddCatery component", () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);
  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });
  test("should show <AddCategory/> correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("should change inputvalue", () => {
    const input = wrapper.find("input");
    const value = "Hello World";
    input.simulate("change", {
      target: {
        value: value,
      },
    });
    expect(wrapper.find("p").text().trim()).toBe(value);
  });
  test("No debe postear la informacion con submit", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });
  test("should call setcategories and clean the inputvalue", () => {
    const input = wrapper.find("input");
    const value = "Hello World";
    input.simulate("change", {
      target: {
        value: value,
      },
    });
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).toHaveBeenCalled();
    expect(wrapper.find("input").prop("value")).toBe("");
  });
});
