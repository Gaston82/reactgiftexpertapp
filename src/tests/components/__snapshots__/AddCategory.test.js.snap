// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AddCatery component should show <AddCategory/> correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
